### PR TITLE
move cli, cache, github-reporter to packages

### PR DESCRIPTION
**Pull request by kamleshchandnani**
This PR moves all the sub projects `cli`, `cache`, `github-reporter` to `packages/*` and enables yarn workspaces.

The new structure also renames the projects to make them more understandable by the name itself. 

Here's how the new naming looks like
`packages/bundlesize-cache-service`
`packages/bundlesize-cli`
`packages/bundlesize-github-reporter-service`

### CODE REVIEW

File: packages/bundlesize-cache-service/index.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ module.exports = async (req, res) => {
       message: `
         Something broke pretty bad!
 
-        Please create an issue with this error message: ${error}
+        Please create an issue at https://github.com/siddharthkp/bundlesize/issues/new with this error message: ${error}
```
Comment (siddharthkp) [inline packages/bundlesize-cache-service/index.js:54]: This link should be bundlesize2 for now

**Comment by siddharthkp** [comment]
> packages/bundlesize-cache-service
> packages/bundlesize-cli
> packages/bundlesize-github-reporter-service

The redundancy in this seems unnecessary as it doesn't match the package name anyways? I'd drop both bundlesize and service from the folder name.

**Comment by kamleshchandnani** [comment]
> > packages/bundlesize-cache-service
> > packages/bundlesize-cli
> > packages/bundlesize-github-reporter-service
> 
> The redundancy in this seems unnecessary as it doesn't match the package name anyways? I'd drop both bundlesize and service from the folder name.

So do you want me to keep it with the same names as they were before?
```
cli
cache
github-reporter
```
since the package.json name doesn't matches even with this dir names

**Comment by siddharthkp** [comment]
> So do you want me to keep it with the same names as they were before?

Yep

**Pull request closed without merging**